api/src/blocks.rs

```diff
@@ -1,66 +1,116 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{context::Context, failpoint::fail_point, metrics::metrics, param::LedgerVersionParam};
-use anyhow::Result;
-use aptos_api_types::{Error, LedgerInfo, Response, TransactionId, U64};
-use warp::{filters::BoxedFilter, Filter, Rejection, Reply};
+use crate::accept_type::AcceptType;
+use crate::context::Context;
+use crate::failpoint::fail_point_poem;
+use crate::response::{
+    BasicErrorWith404, BasicResponse, BasicResponseStatus, BasicResultWith404, InternalError,
+};
+use crate::ApiTags;
+use anyhow::Context as AnyhowContext;
+use aptos_api_types::Block;
+use poem_openapi::param::{Path, Query};
+use poem_openapi::OpenApi;
+use std::sync::Arc;
 
-// GET /blocks/<version>
-pub fn get_block_info(context: Context) -> BoxedFilter<(impl Reply,)> {
-    warp::path!("blocks" / LedgerVersionParam)
-        .and(warp::get())
-        .and(context.filter())
-        .and_then(handle_get_block_info)
-        .with(metrics("get_block_info"))
-        .boxed()
+pub struct BlocksApi {
+    pub context: Arc<Context>,
 }
 
-async fn handle_get_block_info(
-    ledger_version: LedgerVersionParam,
-    context: Context,
-) -> Result<impl Reply, Rejection> {
-    fail_point("endpoint_get_block_info")?;
-    Ok(Block::new(Some(ledger_version), context)?.find_block()?)
-}
+#[OpenApi]
+impl BlocksApi {
+    /// Get blocks by height
```

Comment:
You know we should have probably done this by query parameters instead I realized today, but it's too late